iib/web/api_v1.py

```diff
@@ -37,3 +38,83 @@ def add_bundles():
 
     flask.current_app.logger.debug('Successfully scheduled request %d', request.id)
     return flask.jsonify(request.to_json()), 201
+
+
+@api_v1.route('/builds/<int:request_id>', methods=['PATCH'])
+@login_required
+def patch_request(request_id):
+    """
+    Modify the given request.
+
+    :param int request_id: the request ID from the URL
+    :return: a Flask JSON response
+    :rtype: flask.Response
+    :raise NotFound: if the request is not found
+    :raise ValidationError: if the JSON is invalid
+    """
+    allowed_users = flask.current_app.config['IIB_WORKER_USERNAMES']
+    # current_user.is_authenticated is only ever False when auth is disabled
+    if current_user.is_authenticated and current_user.username not in allowed_users:
+        raise Unauthorized('This API endpoint is restricted to IIB workers')
+
+    payload = flask.request.get_json()
+    if not isinstance(payload, dict):
+        raise ValidationError('The input data must be a JSON object')
+
+    if not payload:
+        raise ValidationError('At least one key must be specified to update the request')
+
+    valid_keys = {
+        'add_arches', 'binary_image_resolved', 'from_index_resolved', 'index_image',
```

Comment:
How about `arches` instead of `add_arches`? `add_arches` is only relevant in the POST request since we are adding to what is currently set on the index image or nothing if there is no index image.

Reply:
sure